asyncstatsd/metrics/datadog.py

```diff
@@ -0,0 +1,42 @@
+from dataclasses import dataclass, field
+from datetime import timedelta
+from typing import Dict
+
+from .basic import GaugeMetricMixin, StatsdMetric, TimingMetricMixin
+
+
+@dataclass
+class DatadogMetric(StatsdMetric):
+    tags: Dict | None = field(default_factory=dict)
+
+    def _build_tags(self) -> str:
+        return ",".join(f"{k}:{v}" for k, v in self.tags.items())  # type: ignore
+
+    def serialize(self) -> str:
+        serialized = super().serialize()
+        tag_string = self._build_tags()
+        serialized = "|#".join(filter(lambda s: s, [serialized, tag_string]))  # type: ignore
+        return serialized
+
+
+@dataclass
+class DatadogCounterMetric(DatadogMetric):
+    unit: str = "c"
+    value: int = 1
+
+
+@dataclass
+class DatadogGaugeMetric(GaugeMetricMixin, DatadogMetric):
+    unit: str = "g"
+    delta: bool = False
+
+
+@dataclass
+class DatadogSetMetric(DatadogMetric):
+    unit: str = "s"
+
+
+@dataclass
+class DatadogTimingMetric(TimingMetricMixin, DatadogMetric):
+    unit: str = "ms"
+    value: int | float | timedelta
```

Comment:
same missing `delta: bool = False` expected by the mixin?

Reply:
nope, see https://github.com/b1r3k/asyncstatsd/pull/1#discussion_r1500608179